Library/Homebrew/diagnostic.rb

```diff
@@ -443,6 +443,32 @@ def check_access_cellar
         EOS
       end
 
+      def check_for_empty_taps
+        taps_dir = HOMEBREW_LIBRARY/"Taps"
+        return unless taps_dir.exist?
+        return unless taps_dir.directory?
+
+        found_empty_dirs = []
+
+        # Warn USER of empty taps_dir children directories
+        taps_dir.children.select{ |c| c.directory? } \
+          .select { |c|
+            if (Dir.entries(c) - %w[. ..]).empty?
+              found_empty_dirs.push(c)
+            end
+          }
+
+        return if found_empty_dirs.empty?
+
+        <<-EOS.undent
+          Found #{found_empty_dirs.length} empty directories in your Taps directory
```

Comment:
Please use `.size` instead if `.length`. It's a minor detail, but using `.size` for collections (like arrays) and `.length` for strings is relatively common in Ruby land and helps avoid confusion.